core/bin/snapshots_creator/src/main.rs

```diff
@@ -0,0 +1,306 @@
+        "Finished storage logs chunk {}/{chunks_count}, step took {elapsed_ms}ms, output stored in {output_filepath}",
+        chunk_id + 1
+    );
+    drop(conn);
+    Ok(output_filepath)
+}
+
+async fn process_factory_deps(
+    blob_store: &dyn ObjectStore,
+    pool: &ConnectionPool,
+    miniblock_number: MiniblockNumber,
+    l1_batch_number: L1BatchNumber,
+) -> anyhow::Result<String> {
+    let latency = METRICS.factory_deps_processing_duration.start();
+    let mut conn = pool.access_storage_tagged("snapshots_creator").await?;
+    let factory_deps = conn
+        .snapshots_creator_dal()
+        .get_all_factory_deps(miniblock_number)
+        .await?;
+    let factory_deps = SnapshotFactoryDependencies { factory_deps };
+    drop(conn);
+    let filename = blob_store
+        .put(l1_batch_number, &factory_deps)
+        .await
+        .context("Error storing factory deps in blob store")?;
+    let output_filepath_prefix = blob_store.get_storage_prefix::<SnapshotFactoryDependencies>();
+    let output_filepath = format!("{output_filepath_prefix}/{filename}");
+    let elapsed_ms = latency.observe().as_millis();
+    tracing::info!(
+        "Finished factory dependencies, step took {elapsed_ms}ms , output stored in {}",
+        output_filepath
+    );
+    Ok(output_filepath)
+}
+
+async fn run(
+    blob_store: Box<dyn ObjectStore>,
+    replica_pool: ConnectionPool,
+    master_pool: ConnectionPool,
+) -> anyhow::Result<()> {
+    let latency = METRICS.snapshot_generation_duration.start();
+
+    let config = SnapshotsCreatorConfig::from_env().context("SnapshotsCreatorConfig::from_env")?;
+
+    let mut conn = replica_pool
+        .access_storage_tagged("snapshots_creator")
+        .await?;
+
+    // we subtract 1 so that after restore, EN node has at least one l1 batch to fetch
+    let l1_batch_number = conn.blocks_dal().get_sealed_l1_batch_number().await? - 1;
+
+    let mut master_conn = master_pool
+        .access_storage_tagged("snapshots_creator")
+        .await?;
+    if master_conn
+        .snapshots_dal()
+        .get_snapshot_metadata(l1_batch_number)
+        .await?
+        .is_some()
+    {
+        tracing::info!("Snapshot for L1 batch number {l1_batch_number} already exists, exiting",);
+        return Ok(());
+    }
+    drop(master_conn);
+
+    // snapshots always
+    let last_miniblock_number_in_batch = conn
+        .blocks_dal()
+        .get_miniblock_range_of_l1_batch(l1_batch_number)
+        .await?
+        .context("Error fetching last miniblock number")?
+        .1;
+    let distinct_storage_logs_keys_count = conn
+        .snapshots_creator_dal()
+        .get_distinct_storage_logs_keys_count(l1_batch_number)
+        .await?;
+
+    drop(conn);
+
+    let chunk_size = config.storage_logs_chunk_size;
+    // we force at least 10 chunks to avoid situations where only one chunk is created in tests
+    let chunks_count = max(10, ceil_div(distinct_storage_logs_keys_count, chunk_size));
```

Comment:
It does not negatively influence the creator. If there are 10 small chunks, even some of them empty that's okay.

Reply:
hmm don't agree with this - while it's technically it doesn't impact correctness, it would look strange at least and incorrect at most if we are generating 10 chunks for a very small dataset. And this can very well happen in the real world - not for Era mainnet, but for some of the hyperchains

Reply:
I agree that's certainly some sort of a tradeoff, but so far it's the best way I see of making sure that we really test generating at least 2 chunks (we don't know, how many storage logs are at the moment the integration test is run)

Reply:
Yeah, the 'root' issue is the non-reproductible test setup. I've been also thinking about chunk number as a configuration parameter, but I think the lesser evil is just to have that hardcoded "10" for now